iterating-over-data/iterating-and-special-cases.md

```diff
@@ -0,0 +1,132 @@
+| `continue` | A keyword that will immediately advance one interation in a loop | -        | "The loop moved on because it saw `continue`"                                                |
+| `range()`  | A Python function that creates a sequence                        | -        | "We used a `for` loop and `range()` in order to iterate through a series of numbers.         |
+
+## `break` Exits a Loop
+
+`break` is a keyword that we can put in a `for` loop. When our code comes across `break`, it will exit the `for` loop entirely.
+
+```python
+for fruit in ["apples", "oranges", "bananas"]:
+    print("I'm in the for loop! The value of fruit is:", fruit)
+    break
+    print("This message will never print, because the for loop saw break before it saw me...")
+```
+
+We can pair this with conditionals (and other logic)!
+
+```python
+print("I'm looking for a four leaf clover in the field...")
+
+for field_item in ["grass", "grass", "more grass", "four-leaf clover", "rocks", "rocks", "more rocks"]:
+    print("...")
+    if field_item == "four-leaf clover":
+        print("I found a four-leaf clover!")
+        print("My hunt is over! I can leave now")
+        break
+    print(f"I found {field_item} in the field...")
+    print("Better keep looking.")
+```
+
+### !callout-info
+
+## for-else
+
+Turns out we can pair a `for` loop with an `else` clause! The `else` clause will run after a `for` loop finishes **if** it did **not** encounter a `break`. The `else` is on the same level of indentation as the `for`.
+
+### !end-callout
+
+```python
+for field_item in ["grass", "grass", "more grass", "rocks", "rocks", "more rocks"]:
+    print("...")
+    if field_item == "four-leaf clover":
+        print("I found a four-leaf clover!")
+        print("My hunt is over! I can leave now")
+        break
+    print(f"I found {field_item} in the field...")
+    print("Better keep looking.")
+else:
+    print("I didn't find a four-leaf clover today.")
+```
+
+## `continue` Advances One Loop
+
+`continue` is another keyword we can put in a `for` loop. When our code sees `continue`, it will immediately move to the next iteration.
+
+```python
+lucky_charms_quantities = {
+    "horseshoe": 4,
+    "four-leaf clover": 0,
+    "lucky coin": 10
+}
+
+for charm, quantity in lucky_charms_quantities.items():
+    if quantity == 0:
+        continue
+    print(f"I found one more {charm}!")
+    lucky_charms_quantities[charm] = quantity + 1
+```
+
+## `range()` Creates a Sequence
+
+Python allows us to create a sequence of numbers using the `range()` function, much like:
+
+- 1, 2, 3
+- 499, 500, 501, 502
+- 3, 6, 9, 12, 15
+- 0, -1, -2, -3.
+
+The function signature of `range` is:
+
+```
+range(start, stop, step)
+```
+
+Where `start` is the number the range should start at, `stop` is the number the range should stop at, and `step` is the amount the range should increment by. `step` is an **optional** argment, and its default value is `1`.
```

Comment:
argument typo.
For this I keep going back and forth about whether I think it would be more clear or more confusing to say that stop is one beyond the number the range should stop at.  As it is, the reader might gloss over the example below and later be surprised, while rephrasing here might make them sit up and take notice, since it sounds strange, but accurate.
```suggestion
Where `start` is the number the range should start at, `stop` is the number the range should stop at, and `step` is the amount the range should increment by. `step` is an **optional** argument, and its default value is `1`.
```
or with the range option
```suggestion
Where `start` is the number the range should start at, `stop` is _one beyond_ the number the range should stop at, and `step` is the amount the range should increment by. `step` is an **optional** argument, and its default value is `1`.
```
and for ultimate pedantry, avoiding dangling prepositions
```suggestion
Where `start` is the number at which the range should start, `stop` is _one beyond_ the number at which the range should stop, and `step` is the amount by which the range should increment. `step` is an **optional** argument, and its default value is `1`.
```

Reply:
```suggestion
Where `start` is the number the range should start at, `stop` is the number the range should stop at, and `step` is the amount the range should increment by. `step` is an **optional** argment, and its default value is `1`.

Where `start` is the number at which the range should start, and `stop` is the number at which the range should stop. `stop` is non-inclusive, which means the `stop` number itself will _not_ be included in the range. `step` is the amount by which the range should increment. `step` is an **optional** argument, and its default value is `1`.
```

Reply:
Bringing the introduction of exclusive up to this point makes the latter mention sound redundant with a little smoothing.  I think it's fine to include here, but take a look at the suggestion for the mention below.